pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -152,3 +167,130 @@ func (c configResource) computeAndReconcileConfigMap(xsite *configuration.XSite)
 	}
 	return err
 }
+
+// addInfinispanConfig installs a reconciler for a specific object.
+// Only events coming from operatorNs/infinispan-operator-config will be processed
+func addInfinispanConfig(mgr manager.Manager) error {
+	r := newInfinispanOperatorConfigReconciler(mgr)
+	// Create a new controller
+	c, err := controller.New("infinispan-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	operatorNS, err := getOperatorNamespace()
+	if err != nil && err != k8sutil.ErrRunLocal {
+		return err
+	}
+	operatorConfigPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetName() == "infinispan-operator-config" && e.MetaNew.GetNamespace() == operatorNS
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetName() == "infinispan-operator-config" && e.Meta.GetNamespace() == operatorNS
+		},
+	}
+	// Watch for changes to global operator config
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, operatorConfigPredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// newInfinispanOperatorConfigReconciler returns a new reconcile.Reconciler
+func newInfinispanOperatorConfigReconciler(mgr manager.Manager) reconcile.Reconciler {
```

Comment:
Is there a need for the function as it's only called once? It's less verbose to just create the struct directly in `addInfinispanConfig`.